File: airbyte-integrations/connectors/destination-mysql-strict-encrypt/src/main/java/io/airbyte/integrations/destination/mysql/MySQLDestinationStrictEncrypt.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.mysql;

import com.fasterxml.jackson.databind.node.ObjectNode;
import io.airbyte.commons.json.Jsons;
import io.airbyte.db.jdbc.JdbcUtils;
import io.airbyte.integrations.base.Destination;
import io.airbyte.integrations.base.IntegrationRunner;
import io.airbyte.integrations.base.spec_modification.SpecModifyingDestination;
import io.airbyte.protocol.models.v0.ConnectorSpecification;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class MySQLDestinationStrictEncrypt extends SpecModifyingDestination implements Destination {

  private static final Logger LOGGER = LoggerFactory.getLogger(MySQLDestinationStrictEncrypt.class);

  public MySQLDestinationStrictEncrypt() {
    super(MySQLDestination.sshWrappedDestination());
  }

  @Override
  public ConnectorSpecification modifySpec(final ConnectorSpecification originalSpec) {
    final ConnectorSpecification spec = Jsons.clone(originalSpec);
    ((ObjectNode) spec.getConnectionSpecification().get("properties")).remove(JdbcUtils.SSL_KEY);
    return spec;
  }

  public static void main(final String[] args) throws Exception {
    final Destination destination = new MySQLDestinationStrictEncrypt();
    LOGGER.info("starting destination: {}", MySQLDestinationStrictEncrypt.class);
    new IntegrationRunner(destination).run(args);
    LOGGER.info("completed destination: {}", MySQLDestinationStrictEncrypt.class);
  }

}

File: airbyte-integrations/connectors/destination-mysql-strict-encrypt/src/main/kotlin/io/airbyte/integrations/destination/mysql/MySQLDestinationStrictEncrypt.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.integrations.destination.mysql

import com.fasterxml.jackson.databind.node.ObjectNode
import io.airbyte.cdk.db.jdbc.JdbcUtils
import io.airbyte.cdk.integrations.base.Destination
import io.airbyte.cdk.integrations.base.IntegrationRunner
import io.airbyte.cdk.integrations.base.spec_modification.SpecModifyingDestination
import io.airbyte.commons.json.Jsons
import io.airbyte.protocol.models.v0.ConnectorSpecification
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class MySQLDestinationStrictEncrypt :
    SpecModifyingDestination(MySQLDestination.sshWrappedDestination()), Destination {
    override fun modifySpec(originalSpec: ConnectorSpecification): ConnectorSpecification {
        val spec: ConnectorSpecification = Jsons.clone(originalSpec)
        (spec.connectionSpecification["properties"] as ObjectNode).remove(JdbcUtils.SSL_KEY)
        return spec
    }

    companion object {
        private val LOGGER: Logger =
            LoggerFactory.getLogger(MySQLDestinationStrictEncrypt::class.java)

        @Throws(Exception::class)
        @JvmStatic
        fun main(args: Array<String>) {
            val destination: Destination = MySQLDestinationStrictEncrypt()
            LOGGER.info("starting destination: {}", MySQLDestinationStrictEncrypt::class.java)
            try {
                IntegrationRunner(destination).run(args)
            } catch (e: Exception) {
                MySQLDestination.handleException(e)
            }
            LOGGER.info("completed destination: {}", MySQLDestinationStrictEncrypt::class.java)
        }
    }
}
